engine-comparison/report-gen/generate-report.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+        "fmt"
+//        "io"
+	"path"
+        "io/ioutil"
+//        "strings"
+        "strconv"
+        "encoding/csv"
+        "os"
+)
+func checkErr(e error){
+	if (e != nil) {
+		fmt.Println("POOP!")
+		os.Exit(1)
+	}
+}
+
+// Removes non-directory elements of any []os.FileInfo (a helper function, for convenience)
+// Makes use of idiomatic Golang return: out_ls is declared and returned implicitly
+func onlyDirectories(potential_files []os.FileInfo) (out_ls []os.FileInfo) {
+	for _, fd := range potential_files {
+		if fd.IsDir() {
+			out_ls = append(out_ls, fd)
+		}
+	}
+	return
+}
+
+// Extend "records" matrix to have rows until time "desired_time"
+// Return: Extended version of record
+func extendRecordsToTime(records [][]string, desired_time int, recordCols int) ([][]string)  {
+	lenr := len(records)
+	// records[1] stores cycle [1], as records[0] is column names
+	for j := lenr; j < desired_time + 1; j++ {
+		records = append(records, make([]string, recordCols))
+		records[j][0] = strconv.Itoa(j)
+		for k := 1; k < recordCols; k++ {
+			records[j][k] = ""
```

Comment:
Redundant.  I believe make already sets default empty strings.

Reply:
I had a couple issues with `append` because strings are already lists, but you're definitely right about this, thanks